api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -911,3 +907,17 @@ message UpstreamConnectionOptions {
   // If set then set SO_KEEPALIVE on the socket to enable TCP Keepalives.
   core.v3.TcpKeepalive tcp_keepalive = 1;
 }
+
+message TrackOptionalClusterStats {
```

Comment:
imho `Optional` can be dropped from name, as tracking of stats is turned on/off with boolean flags, it implies that it can be specified or not.

Reply:
will update.

Reply:
Need to improve my nomenclature decisions :100: